Add example documentation to show MLeap functionality for SparkML flavor (#7657)
mlflow/mlflow: docs/source/models.rst

pipelinemodel#pyspark.ml.Pipeline>`_ to any production environment supported by MLflow
(SageMaker, AzureML, etc).

.. note::

>     Note that when the ``sample_input`` parameter is provided to ``log_model()`` or 

Reviewer: Let's make it VERY clear that this is what this is for. It's for saving a SparkML model to be used in MLeap. Not for using as a SparkML model. 

Let's put a ```.. warning:: `` block within this note that clearly explains this.

Author: I don't quite understand what the play is here. Per the documentation [mlflow.mleap](https://mlflow.org/docs/latest/python_api/mlflow.mleap.html#module-mlflow.mleap) we can't extract an MLeap model using the Python API. Is the expectation that the user only uses the Java API? I feel like that's a major turn off

Author: also technically you can do both. You can extract as a spark model or you can also use the MLeap artifact

Author: Also it seems like you are able to load a model using the` mlflow.spark` flavor even if it was saved with `sample_input`

Reviewer: 1. Yep you'll have to make that clear as well. MLeap is JVM-only (SparkML Scala APIs). It is what it is.
2. Loading as Spark with MLeap applied - good point. Let's make that clear about how you can still load as SparkML but if you don't specify this argument in saving, you can't load as MLeap since it won't understand how to load the flavor correctly.

>     ``save_model()``, the Spark model is automatically saved with using the ``mleap`` flavor

Reviewer: ```suggestion
    ``save_model()``, the Spark model is automatically saved as an ``mleap`` flavor
```
